oauthproxy.go

```diff
@@ -519,6 +519,11 @@ func (p *OAuthProxy) ManualSignIn(rw http.ResponseWriter, req *http.Request) (st
 // GetRedirect reads the query parameter to get the URL to redirect clients to
 // once authenticated with the OAuthProxy
 func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error) {
+	if p.SkipProviderButton {
+		redirect = req.RequestURI
+		return
```

Comment:
this should probably still check `IsValidRedirect()` ... actually, I'd expect the below code to handle this case:

```go
	// no form and no header -> blank redirect -> invalid
	if !p.IsValidRedirect(redirect) {
		redirect = req.URL.Path
```

maybe you are running into some other problem ...

Reply:
I agree with @ploxiln, I don't think this is needed given the code that he highlighted. Perhaps there is some other issue causing your problem as the `SkipProviderButton` shouldn't be affecting your redirect. Have you tried adding some debug logging in here to check the redirect as it's being set?